presto-postgresql/src/test/java/io/prestosql/plugin/postgresql/TestPostgreSqlDistributedQueries.java

```diff
@@ -33,6 +32,10 @@ protected QueryRunner createQueryRunner()
             throws Exception
     {
         postgreSqlServer = new TestingPostgreSqlServer();
+        registerResource(() -> {
+            postgreSqlServer.close();
+            postgreSqlServer = null;
```

Comment:
`registerResource` is misleading, as you're registering a callback

(the declaration looks legit, but the call site like this looks weird)

Reply:
I was thinking also about `closeAfterClass`? WDYT?